hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -560,6 +560,25 @@ public class HoodieWriteConfig extends HoodieConfig {
       .withDocumentation("When table is upgraded from pre 0.12 to 0.12, we check for \"default\" partition and fail if found one. "
           + "Users are expected to rewrite the data in those partitions. Enabling this config will bypass this validation");
 
+  public static final ConfigProperty<String> RECORD_SIZE_DYNAMIC_SAMPLING_MAXNUM = ConfigProperty
```

Comment:
@weimingdiit let's reduce number of user-facing configuration we provide as part of this feature: IMO we should have just 2:

 - Whether dynamic sampling is enabled (by default we should switch this to true)
 - Total number of records to be sampled (this should be deliberately be static, to avoid variation with the batch size if, say, one day we ingest 10Gb batch and in a month we start ingesting 100Gb batches)